src/edu/psu/sweng500/team8/gui/CustomKeyListener.java

```diff
@@ -24,7 +24,10 @@ public CustomKeyListener(Cell cell, GameSession gameSession) {
 
 	public void keyTyped(java.awt.event.KeyEvent keyEvent) {
 		
-		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar()) {
+		
+		
+		if (KeyEvent.VK_BACK_SPACE == keyEvent.getKeyChar() || KeyEvent.VK_DELETE == keyEvent.getKeyChar()) {
+			
 			gameSession.enterNumber(this.cell, 0);
 		} else {
 			gameSession.enterNumber(this.cell, Integer.parseInt(String.valueOf(keyEvent.getKeyChar())));

```

Comment:
David's logic checks the key to ensure it is a valid integer. If this event handler is hit before David's, or if both event handlers are called, this will throw an exception if an invalid character is entered.